packages/core/src/database/update-where.ts

```diff
@@ -35,6 +35,7 @@ export const buildUpdateWhere: BuildUpdateWhere = <Schema extends SchemaLike>(
   const isKeyOfSchema = isKeyOf(schema);
   const connectKeyValueWithEqualSign = (data: Partial<Schema>) =>
     Object.entries(data)
+      .filter(([_, value]) => value !== undefined)
```

Comment:
In typeorm, `undefined` value will be generated as `IS NULL` in sql.
It can avoid unexpected lost of where contitions, which may cause data leaking. (undefined may appear unexpectly)
https://github.com/typeorm/typeorm/issues/737